src/libraries/System.Net.Quic/src/Interop/MsQuicNativeMethods.cs

```diff
@@ -435,7 +435,7 @@ internal struct SOCKADDR_INET
         internal delegate uint StreamCallbackDelegate(
             IntPtr stream,
             IntPtr context,
-            StreamEvent streamEvent);
+            ref StreamEvent streamEvent);
```

Comment:
Should this be `in`? Are we mutating the struct?

Reply:
I think they should be ref as some events can be modified. The parameter in native is defined as inout. https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicListener.cs#L149. Though we don't do it currently, we could modify the security configuration in the listener event to set the security configuration.